src/models/repo_demo.ts

```diff
@@ -353,63 +345,90 @@ export function editUser(new_user:Person){
 /**
  * Function returns all orders from db
  */
-export function getAllOrders():Order[]{
+export function getAllOrders(): Order[] {
     return orders;
 }
 
 /**
  * Function returns user data
  */
-export function getOrdersbyUserID(id: number):Order[]{
+export function getOrdersbyUserID(id: number): Order[] {
     return orders.filter(o => o.UserID === id)
 }
 
 /**
  * Function adds order to db
  */
-export function pushNewOrder(o: Order){
+export function pushNewOrder(o: Order) {
     orders.push(o);
     settings.number_of_orders += 1
 }
 
 /**
  * Function returns all products from db
  */
-export function getAllProducts(): Product[]
-{
+export function getAllProducts(): Product[] {
     return products;
 }
 
 /**
  * Function returns one product(by id) from db
  */
-export function getProductbyID(id:number):Product|null{
-    const res =  products.find(u => u.ID === id);
-    if(res !== undefined)
-    {
+export function getProductbyID(id: number): Product | null {
+    const res = products.find(u => u.ID === id);
+    if (res !== undefined) {
         return res;
     }
-    else
-    {
+    else {
         return null;
     }
 }
 
 /**
  * Function adds one new product to db.
  */
-export function pushNewProduct(prod: Product){
+export function pushNewProduct(prod: Product) {
     products.push(prod);
-    settings.number_of_products+=1;
+    settings.number_of_products += 1;
 }
 
 /**
  * Function adds one new product to db.
  */
-export function editProduct(prod: Product){
+export function editProduct(prod: Product) {
     let index = products.findIndex(p => p.ID === prod.ID)
-    if(index !== -1)
-    {
-        products[index]=prod;
+    if (index !== -1) {
+        products[index] = prod;
+    }
+}
+
+/**
+ * Function deletes one product from db.
+ */
+export function deleteProduct(prod: Number) {
```

Comment:
Jak dodamy obsługę bazy danych trzeba będzie ustalić co robimy z produktami które już są w jakimś `Pending` zamówieniu

Reply:
Kurcze to bardzo ciężka sprawa będzie. Edytowanie istniejącego produktu rozwala nam naszą logikę historii zamówień.